pandas/core/algorithms.py

```diff
@@ -1658,8 +1658,25 @@ def take(arr, indices, axis: int = 0, allow_fill: bool = False, fill_value=None)
     return result
 
 
+# TODO: can we de-duplicate with something in dtypes.missing?
+def _get_default_fill_value(dtype, fill_value):
+    if fill_value is lib.no_default:
+        if is_extension_array_dtype(dtype):
+            fill_value = dtype.na_value
```

Comment:
na_value_for_dtype

Reply:
thats what i thought too, but its behavior is slightly different